include/RAJA/util/Layout.hpp

```diff
@@ -209,15 +216,19 @@ struct LayoutBase_impl<camp::idx_seq<RangeInts...>, IdxLin, StrideOneDim> {
 #if defined(RAJA_BOUNDS_CHECK_INTERNAL)
     IdxLin totSize{1};
     for(size_t i=0; i<n_dims; ++i) {totSize *= sizes[i];};
-    if(totSize > 0 && (linear_index < 0 || linear_index >= totSize)) {
+    if((totSize > 0 || !allow_projection) &&
+       (linear_index < 0 || linear_index >= totSize)) {
       printf("Error! Linear index %ld is not within bounds [0, %ld]. \n",
              static_cast<long int>(linear_index), static_cast<long int>(totSize-1));
       RAJA_ABORT_OR_THROW("Out of bounds error \n");
      }
 #endif
 
-    camp::sink((indices = (camp::decay<Indices>)((linear_index / inv_strides[RangeInts]) %
-                                   inv_mods[RangeInts]))...);
+    camp::sink((indices = (camp::decay<Indices>)(
+      ( RangeInts==stride_one_dim ?   // Is this dimension stride-one?
+          (linear_index % inv_mods[RangeInts]) :  // it's stride one, so don't bother with divide
+          ((linear_index / inv_strides[RangeInts]) % inv_mods[RangeInts]) // it's not stride one
+      ) ))...);
```

Comment:
This bit makes me slightly nervous.  Any performance numbers on whether this is going to introduce an issue at runtime?

Reply:
I have not done any performance checks yet, are you concerned the compiler will fail to vectorize if the operations are not symmetric?

Reply:
I'm concerned that the ternary might not get optimized out and turn into a conditional branch or a conditional store.  It *should* get optimized out in constant propagation, but this is a place where performance could tank pretty hard if it doesn't.

Reply:
We do this to handle the stride_one index as well.

Reply:
I split out the calculations into classes and partially specialized so there is no ternary anymore.